evaluator/evaluate_search.go

```diff
@@ -141,7 +141,24 @@ eventMatcher:
 
 func (rule *RuleEvaluator) getMatcherValues(ctx context.Context, matcher sigma.FieldMatcher) ([]string, error) {
 	matcherValues := []string{}
-	for _, value := range matcher.Values {
+	for _, abstractValue := range matcher.Values {
+		value := ""
+
+		switch abstractValue := abstractValue.(type) {
+		case string:
+			value = abstractValue
+		case int:
+			value = fmt.Sprintf("%v", abstractValue)
+		case float32:
+			value = fmt.Sprintf("%v", abstractValue)
+		case float64:
+			value = fmt.Sprintf("%v", abstractValue)
+		case bool:
+			value = fmt.Sprintf("%v", abstractValue)
```

Comment:
```suggestion
		case int, float32, float64, bool:
			value = fmt.Sprintf("%v", abstractValue)
```

Reply:
I feel really dumb that I didn't know this was a valid golang syntax. Thank you so much. lol